client/ayon_core/pipeline/create/product_name.py

```diff
@@ -159,14 +180,29 @@ def get_product_name(
         task_short = task_types_by_name.get(task_type, {}).get("shortName")
         task_value["short"] = task_short
 
-    fill_pairs = {
+    # look what we have to do to make mypy happy. We should stop using
+    # those undefined dict based types.
+    product: dict[str, str] = {
+        "type": product_type,
+        "baseType": product_base_type
+    }
+    if not product_base_type and "{product[basetype]}" in template.lower():
+        product["baseType"] = product_type
+        warn(
+            "You have Product base type in product name template, "
+            "but it is not provided by the creator, please update your "
+            "creation code to include it. It will be required in "
+            "the future.",
+            DeprecationWarning,
+            stacklevel=2)
```

Comment:
```suggestion
    if not product_base_type and "{product[basetype]}" in template.lower():
        warn(
            "You have Product base type in product name template, "
            "but it is not provided by the creator, please update your "
            "creation code to include it. It will be required in "
            "the future.",
            DeprecationWarning,
            stacklevel=2)
        
    product: dict[str, str] = {
        "type": product_type,
        "baseType": product_base_type or product_type,
    }
```